packages/@aws-cdk/core/lib/runtime.ts

```diff
@@ -55,6 +59,110 @@ export function cfnTagToCloudFormation(x: any): any {
   };
 }
 
+export function cloudFormationTagToCdkTag(tag: any): CfnTag {
+  return tag == null
+    ? { } as any // break the type system - this should be detected at runtime by a tag validator
+    : {
+      key: tag.Key,
+      value: tag.Value,
+    };
+}
+
+// nothing to do - if 'value' is not a boolean or a Token,
+// a validator should report that at runtime
+export const booleanFromCloudFormation: (value: any) => boolean | IResolvable = identity;
+
+// nothing to for any but return it
+export const anyFromCloudFormation = identity;
+
+export function dateFromCloudFormation(value: any): Date | IResolvable {
+  // if the date is a deploy-time value, just return it
+  if (isResolvableObject(value)) {
+    return value;
+  }
+
+  // if the date has been given as a string, convert it
+  if (typeof value === 'string') {
+    return new Date(value);
+  }
+
+  // all other cases - just return the value,
+  // if it's not a Date, a validator should catch it
+  return value;
+}
+
+export function stringFromCloudFormation(value: any): string {
+  // if the string is a deploy-time value, serialize it to a Token
+  if (isResolvableObject(value)) {
+    return value.toString();
+  }
+
+  // in all other cases, just return the input,
+  // and let a validator handle it if it's not a string
+  return value;
+}
+
+export function numberFromCloudFormation(value: any): number {
+  // if the string is a deploy-time value, serialize it to a Token
+  if (isResolvableObject(value)) {
+    return Token.asNumber(value);
+  }
+
+  // in all other cases, just return the input,
+  // and let a validator handle it if it's not a number
+  return value;
+}
+
+export function stringArrayFromCloudFormation(value: any): string[] {
+  // if the array is a deploy-time value, serialize it to a Token
+  if (isResolvableObject(value)) {
+    return Token.asList(value);
+  }
+
+  // in all other cases, delegate to the standard mapping logic
+  return cloudFormationArrayToCdkArray(value, stringFromCloudFormation);
+}
+
+export function cloudFormationArrayToCdkArray<T>(value: any, mapper: (arg: any) => T): T[] {
```

Comment:
Awww why did we switch from `xxxFromYyy` to `yyyToXxx` here?

Reply:
Sure. Changed all the names to be `<sth>FromCloudFormation`.